src/ucs/sys/ucs_netlink.c

```diff
@@ -0,0 +1,119 @@
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "ucs_netlink.h"
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <arpa/inet.h>
+#include <net/if.h>
+#include <linux/netlink.h>
+#include <ucs/type/status.h>
+#include <ucs/debug/log.h>
+
+ucs_status_t netlink_socket_create(struct netlink_socket *nl_sock, int protocol)
+{
+    int fd;
+
+    fd = socket(AF_NETLINK, SOCK_RAW, protocol);
+    if (fd < 0) {
+        ucs_diag("failed to create a socket\n");
+        return UCS_ERR_IO_ERROR;
+    }
```

Comment:
use `ucs_socket_create` instead.

Reply:
The problem is that I want ucs_netlink user to be able to specify the netlink protocol to use (NETLINK_ROUTE, NETLINK_FIREWALL, NETLINK_RDMA, etc.). Specifically in its current use it doesn't matter because NETLINK_ROUTE is equal to 0 (which is the default value defined in `ucs_socket_create`) but in other cases I want to enable it..

Reply:
right, so need to add `protocol` argument to `ucs_socket_create`